create ImageCollector play

AhnafAhamed opened the pull request:
# Description

Created the Image Collector Play.

Fixes #645 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

atapas commented:
@murtuzaalisurti plz review

murtuzaalisurti commented:
> @murtuzaalisurti plz review

yeah

murtuzaalisurti commented on src/plays/image-collector/styles.css:
```diff
@@ -0,0 +1,51 @@
+
+.input-url {
+  padding: 8px 16px;
+  width: 400px;
```
please make the input element responsive by using relative css units

murtuzaalisurti reviewed (changes requested):
there are some changes that need to be made

koustov reviewed (changes requested):
Early feedback. Need to maintain the image aspect ratio
![image](https://user-images.githubusercontent.com/7145967/194723032-bc941034-d27c-4089-b83d-9fa4f635f14a.png)

AhnafAhamed commented:
Thanks for the feedback @murtuzaalisurti @koustov. I have added the requested changes. Thank You

murtuzaalisurti commented:
@AhnafAhamed the aspect ratio is still not correct for some images as mentioned by @koustov, maybe you can do `background-size: cover`.

here's some reference https://css-tricks.com/almanac/properties/b/background-size/

AhnafAhamed commented:
> @AhnafAhamed the aspect ratio is still not correct for some images as mentioned by @koustov, maybe you can do `background-size: cover`.
> 
> here's some reference https://css-tricks.com/almanac/properties/b/background-size/

I have added `object-fit: cover;` to the image which is similar to `background-size` property. Do you want me to make this image a background-image and add the `background-size: cover;` property? Thanks

murtuzaalisurti commented:
@atapas the vercel preview is not updating

murtuzaalisurti reviewed (approved):
looks fine

atapas commented:
@AhnafAhamed Please take care of @koustov's comments.

AhnafAhamed commented:
> @AhnafAhamed Please take care of @koustov's comments.

Hi @atapas 

I have already fixed it and @murtuzaalisurti also said it's fine. 

Below you can see I have added different images and there is no problem with the aspect ratio.


![image](https://user-images.githubusercontent.com/42427928/196397824-30cc213c-f2d4-47e0-b131-4e1ed4e51d85.png)

atapas commented:
Thanks @AhnafAhamed 

@koustov please review

koustov commented:
I still have some observation

1. Images are a bit scattered over the container. I mean these two images are way apart not properly center-aligned. 
2. Also the landscape images are cropped. Is it by design? 
3. The boxes should have bit of borders. Notice the PNG image borders are kind of vanished with the background 

![image](https://user-images.githubusercontent.com/7145967/196510259-107b7b03-2436-4c7d-a9ef-e1df1d59fe9d.png)

![image](https://user-images.githubusercontent.com/7145967/196510825-b4970e37-468f-4237-90a3-ef390c8af5e4.png)

koustov commented on src/plays/image-collector/readme.md:
```diff
@@ -0,0 +1,27 @@
+
+## Consideration
+
+Update all considerations(if any)
```
Remove the section if not applicable

koustov commented on src/plays/image-collector/readme.md:
```diff
@@ -0,0 +1,27 @@
+
+## Implementation Details
+
+Update your implementation idea and details here
```
Give implementation details for your play

koustov commented on src/plays/image-collector/readme.md:
```diff
@@ -0,0 +1,27 @@
+
+## Resources
+
+Update external resources(if any)
```
Remove the section if not applicable

AhnafAhamed commented:
Hi @koustov 

I have made improvements base on your suggestions, please review them.

Thank You for the feedback.

AhnafAhamed commented:
Hi @koustov any updates on this?

murtuzaalisurti commented on src/plays/image-collector/ImageCollector.js:
```diff
@@ -0,0 +1,54 @@
+function ImageCollector(props) {
+  const [urlInput, setUrlInput] = useState("");
+  const [images, setImages] = useState([]);
+  // const images = useRef([]);
```
can you remove this comment if it's not required?

AhnafAhamed replied:
@murtuzaalisurti removed it

murtuzaalisurti replied:
great, looks good

koustov reviewed (approved):
lgtm

@murtuzaalisurti is there anything pending from your end?

murtuzaalisurti commented:
> lgtm
> 
> @murtuzaalisurti is there anything pending from your end?

nope, all clear

murtuzaalisurti reviewed (approved):
looks good
